// Code generated by mockery v2.53.3. DO NOT EDIT.

package mock

import (
	context "context"

	flow "github.com/onflow/flow-go/model/flow"

	mock "github.com/stretchr/testify/mock"
)

// TxErrorMessageProvider is an autogenerated mock type for the TxErrorMessageProvider type
type TxErrorMessageProvider struct {
	mock.Mock
}

// LookupErrorMessageByIndex provides a mock function with given fields: ctx, blockID, height, index
func (_m *TxErrorMessageProvider) LookupErrorMessageByIndex(ctx context.Context, blockID flow.Identifier, height uint64, index uint32) (string, error) {
	ret := _m.Called(ctx, blockID, height, index)

	if len(ret) == 0 {
		panic("no return value specified for ErrorMessageByIndex")
	}

	var r0 string
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, flow.Identifier, uint64, uint32) (string, error)); ok {
		return rf(ctx, blockID, height, index)
	}
	if rf, ok := ret.Get(0).(func(context.Context, flow.Identifier, uint64, uint32) string); ok {
		r0 = rf(ctx, blockID, height, index)
	} else {
		r0 = ret.Get(0).(string)
	}

	if rf, ok := ret.Get(1).(func(context.Context, flow.Identifier, uint64, uint32) error); ok {
		r1 = rf(ctx, blockID, height, index)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// LookupErrorMessageByTransactionID provides a mock function with given fields: ctx, blockID, height, transactionID
func (_m *TxErrorMessageProvider) LookupErrorMessageByTransactionID(ctx context.Context, blockID flow.Identifier, height uint64, transactionID flow.Identifier) (string, error) {
	ret := _m.Called(ctx, blockID, height, transactionID)

	if len(ret) == 0 {
		panic("no return value specified for ErrorMessageByTransactionID")
	}

	var r0 string
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, flow.Identifier, uint64, flow.Identifier) (string, error)); ok {
		return rf(ctx, blockID, height, transactionID)
	}
	if rf, ok := ret.Get(0).(func(context.Context, flow.Identifier, uint64, flow.Identifier) string); ok {
		r0 = rf(ctx, blockID, height, transactionID)
	} else {
		r0 = ret.Get(0).(string)
	}

	if rf, ok := ret.Get(1).(func(context.Context, flow.Identifier, uint64, flow.Identifier) error); ok {
		r1 = rf(ctx, blockID, height, transactionID)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// LookupErrorMessagesByBlockID provides a mock function with given fields: ctx, blockID, height
func (_m *TxErrorMessageProvider) LookupErrorMessagesByBlockID(ctx context.Context, blockID flow.Identifier, height uint64) (map[flow.Identifier]string, error) {
	ret := _m.Called(ctx, blockID, height)

	if len(ret) == 0 {
		panic("no return value specified for ErrorMessagesByBlockID")
	}

	var r0 map[flow.Identifier]string
	var r1 error
	if rf, ok := ret.Get(0).(func(context.Context, flow.Identifier, uint64) (map[flow.Identifier]string, error)); ok {
		return rf(ctx, blockID, height)
	}
	if rf, ok := ret.Get(0).(func(context.Context, flow.Identifier, uint64) map[flow.Identifier]string); ok {
		r0 = rf(ctx, blockID, height)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(map[flow.Identifier]string)
		}
	}

	if rf, ok := ret.Get(1).(func(context.Context, flow.Identifier, uint64) error); ok {
		r1 = rf(ctx, blockID, height)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// NewTxErrorMessageProvider creates a new instance of TxErrorMessageProvider. It also registers a testing interface on the mock and a cleanup function to assert the mocks expectations.
// The first argument is typically a *testing.T value.
func NewTxErrorMessageProvider(t interface {
	mock.TestingT
	Cleanup(func())
}) *TxErrorMessageProvider {
	mock := &TxErrorMessageProvider{}
	mock.Mock.Test(t)

	t.Cleanup(func() { mock.AssertExpectations(t) })

	return mock
}
